action.yml

```diff
@@ -110,6 +110,13 @@ runs:
         Get-ChildItem Env: | ForEach-Object { echo "$($_.Name)=$($_.Value)" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append }
       shell: pwsh
 
+    - name: Upload Error Logs
+      if: runner.os == 'Windows' && failure()
```

Comment:
I think this could be scoped better by adding `id: install-swift` to the `Install Swift ${{ inputs.tag }}` step and then changing the predicate here to be
```
if: failure() && steps.install-swift.outcome == 'failure'
```
to avoid missing file errors on `${env:Temp}/install*.log*` if, say, the failure is from the installer failing to download.

Reply:
Thats a good idea, I've incorporated that into the change.